src/observation.jl

```diff
@@ -40,13 +40,13 @@ Every author behind some custom data container can make this
 decision themselves.
 The output should be consistent when `idx` is a scalar vs vector.
 
-See also [`getobs!`](@ref) and [`numobs`](@ref) 
+See also [`getobs!`](@ref) and [`numobs`](@ref)
 """
 function getobs end
 
 # Generic Fallbacks
 getobs(data) = data
-getobs(data, idx) = data[idx]
+getobs(data, idx) = istable(data) ? collect(rows(data))[idx] : data[idx]
```

Comment:
Collecting all rows is too expensive, let's wait for the outcome of the discussion here
https://github.com/JuliaData/Tables.jl/pull/278#issuecomment-1132706343